System/Primitive.cpp

```diff
@@ -52,6 +52,70 @@ namespace RTE {
 		}
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ArcPrimitive::Draw(BITMAP *drawScreen, Vector targetPos) {
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart = m_StartPos - targetPos;
+			if (m_Thickness > 1) {
+				for (short i = 0; i < m_Thickness; i++) {
+					arc(drawScreen, drawStart.m_X, drawStart.m_Y, ftofix(GetAllegroAngle(-m_EndAngle)), ftofix(GetAllegroAngle(-m_StartAngle)), (m_Radius - (m_Thickness / 2)) + i, m_Color);
```

Comment:
Can you explain to me why you're starting at -EndAngle and ending at -StartAngle? I don't understand this.